src/porepy/numerics/ad/surrogate_operator.py

```diff
@@ -1006,5 +1045,25 @@ def progress_iterate_derivatives_on_grid(
             iterate_index=0,
         )
         data = self._data_of(grid)
-        pp.shift_solution_values(self._name_derivatives, data, pp.ITERATE_SOLUTIONS)
+        pp.shift_solution_values(
+            self._name_derivatives, data, pp.ITERATE_SOLUTIONS, max_index=depth
+        )
         pp.set_solution_values(self._name_derivatives, values, data, iterate_index=0)
+
+    def shift_iterate_values_on_grids(
```

Comment:
I asked the same question above: What is the difference between this method and `progress_iterate_...`?

Reply:
I explained it there https://github.com/pmgbergen/porepy/pull/1186#discussion_r1637587228

I added this ``shift_...`` method purely for convenience to not have the code of this method at multiple places in the CF solution strategy. Could be seen as a bit exaggerated, but I tried to avoid repetitive code as much as possible.

Reply:
Removing shift methods, considered a redundant utility function